mongo/index_view.go

```diff
@@ -435,6 +462,20 @@ func (iv IndexView) DropOne(ctx context.Context, name string, opts ...*options.D
 	return iv.drop(ctx, name, opts...)
 }
 
+// DropKeyOne executes a dropIndexes operation to drop an index on the collection. If the operation succeeds, this returns
+// a BSON document in the form {nIndexesWas: <int32>}. The "nIndexesWas" field in the response contains the number of
+// indexes that existed prior to the drop.
+//
+// The key parameter should be the keySpecDocument of the index to drop.
+func (iv IndexView) DropKeyOne(ctx context.Context, keySpecDocument interface{}, opts ...*options.DropIndexesOptions) (bson.Raw, error) {
```

Comment:
`DropKeyOne` is a confusing name, suggest renaming to something more clear like `DropByKey()` or `DropWithKey()`.